src/Bicep.Core/Analyzers/Linter/Rules/SecureParamsInNestedDeploymentsRule.cs

```diff
@@ -0,0 +1,141 @@
+        //
+        // These nested deployments will evaluate expressions prior to being sent to the deployment engine.This means that all
+        // properties are in the request in an evaluated state (clear text) and persisted on the deployment object.
+
+        public new const string Code = "secure-params-in-nested-deploy";
+
+        public SecureParamsInNestedDeploymentsRule() : base(
+            code: Code,
+            description: CoreResources.SecureParamsInNestedDeployRule_Description,
+            docUri: new Uri($"https://aka.ms/bicep/linter/{Code}"),
+            diagnosticLevel: DiagnosticLevel.Warning)
+        { }
+
+        public override string FormatMessage(params object[] values)
+        {
+            string problem = (string)values[0];
+            return $"{problem} {CoreResources.SecureParamsInNestedDeployRule_Solution}";
+        }
+
+        override public IEnumerable<IDiagnostic> AnalyzeInternal(SemanticModel model)
+        {
+            foreach (ResourceSymbol resource in model.Root.ResourceDeclarations)
+            {
+                if (GetPropertieIfOuterScopedDeployment(resource) is ObjectSyntax propertiesObject)
+                {
+                    var templateObject = propertiesObject.TryGetPropertyByName("template")?.Value as ObjectSyntax;
+                    if (templateObject is not null)
+                    {
+                        // Look for secure params
+                        var secureParams = FindSecureParametersReferencedInNestedDeployment(model, resource);
+                        if (secureParams.Any())
+                        {
+                            string secureParamsAsString = String.Join(", ", secureParams.Select(p => $"'{p.Name}'"));
+                            var message = string.Format(
+                                CoreResources.SecureParamsInNestedDeployRule_Message_SecureParams,
+                                resource.Name,
+                                secureParamsAsString);
+                            yield return CreateDiagnosticForSpan(resource.NameSyntax.Span, message);
+                        }
+
+                        // Look for list* functions
+                        var listFunctionReference = FindListFunctionReferencesInSyntaxTree(model, resource.DeclaringSyntax).FirstOrDefault();
+                        if (listFunctionReference is not null)
+                        {
+                            var message = string.Format(
+                                CoreResources.SecureParamsInNestedDeployRule_Message_ListFunction,
+                                resource.Name,
+                                listFunctionReference.ToText());
+                            yield return CreateDiagnosticForSpan(resource.NameSyntax.Span, message);
+                        }
+                    }
+                }
+            }
+        }
+
+        private static ObjectSyntax? GetPropertieIfOuterScopedDeployment(ResourceSymbol resource)
```

Comment:
nit: Can be refactored to below to improve readability:
```
        private static ObjectSyntax? GetPropertieIfOuterScopedDeployment(ResourceSymbol resource)
        {
            if (resource.TryGetResourceType() is not ResourceType resourceType)
            {
                return null;
            }

            if (!resourceType.TypeReference.FormatType().Equals("microsoft.resources/deployments", LanguageConstants.ResourceTypeComparison))
            {
                return null;
            }

            if (resource.TryGetBodyProperty(LanguageConstants.ResourcePropertiesPropertyName)?.Value is not ObjectSyntax propertiesObject)
            {
                return null;
            }

            // It's a valid deployment resource
            if (propertiesObject is not null && propertiesObject.TryGetPropertyByNameRecursive(new[] { "expressionEvaluationOptions", "scope" })
                ?.Value
                is StringSyntax scope)
            {
                if (scope.TryGetLiteralValue()?.Equals("inner", StringComparison.InvariantCultureIgnoreCase) == true)
                {
                    // Scope is explicitly marked as "inner"
                    return null;
                }
            }

            // Default scope is outer if not explicitly marked as inner
            return propertiesObject;
        }
```

Reply:
Thanks.  I had assumed the pattern variable would be out of scope outside of the if/then.